### PR TITLE
feat: Add pre-commit framework

### LINKED ISSUE #681: Use Pre-commit libary in the project
Hi everyone :wave:  
Thanks for this awesome Project!

When setting up my development environment (i.e. running `make dev-env`), I get the following error
```
make dev-env                                                                                                                                                                                                                           
Found system python version: 3.10

...

Successfully installed pelorus-0.0.0
⎇ Configuring git to ignore certain revs for annotations
if [ "" != ".git-blame-ignore-revs" ]; then \
	git config blame.ignoreRevsFile .git-blame-ignore-revs; \
fi
./scripts/setup-pre-commit-hook
⎇ setting up pre-commit hook
warning: pre-commit hook exists, should be symlink to .git/hooks/pre-commit
make: *** [Makefile:76: .git/hooks/pre-commit] Erro 1
```
I believe this errors occurs because I have configured [pre-commit library](https://pre-commit.com/) to automatically setup it self when cloning a repository ([more info here](https://pre-commit.com/#automatically-enabling-pre-commit-on-repositories)).

Can I setup Pelorus' pre-commit script without touching my pre-commit library configuration?

And there is a reason Pelorus does not use the pre-commit library?

### LINKED ISSUE #692: Pre-commit checks don't support forking
It seems that when a user tries to make a commit, the pre-commit chart linting step makes an assumption that the `konveyor/pelorus` remote as set as `origin` in the users local git setup. If its not, the lint step will fail with the following error:

```bash
./scripts/install_dev_tools -v .venv -c ct && \
. .venv/bin/activate && \
ct lint --config ct.yaml
Cleaning up temporary files
Linting charts...
------------------------------------------------------------------------------------------------------------------------
No chart changes detected.
------------------------------------------------------------------------------------------------------------------------
Error: failed linting charts: failed identifying charts to process: failed running process: exit status 128
failed linting charts: failed identifying charts to process: failed running process: exit status 128
make: *** [Makefile:221: chart-lint] Error 1
🐚 📋 Linting shell scripts with shellcheck
/bin/shellcheck demo/demo-tekton.sh scripts/create_release_pr scripts/install_dev_tools scripts/pre-commit scripts/run-mockoon-tests scripts/setup-pre-commit-hook scripts/run-pelorus-e2e-tests
make: Target 'lint' not remade because of errors.


Linting failed. Check the above output for hints.


If you must, use git commit --no-verify to bypass these checks.
```

This setting seems to discourage repository forking, which I don't think we want. Ideally this setting could either dynamically figure out the right remote to compare against for chart linting, or allow the user to set that up somehow, OR at worst default to expect the user to be using a fork and have `konveyor/pelorus` set as `upstream` instead of `origin`.

### PR DESCRIPTION
**Pull request by mateusoliveira43**
## Describe the behavior changes introduced in this PR

Adds [pre-commit framework](https://pre-commit.com/) to Pelorus.

## Linked Issues

resolves #681
resolves #692

related to #780

## Testing Instructions

Run `make dev-env` (this will run `pre-commit install` and setup pre-commit to you). Then, run `pre-commit run --all-files` to run pre-commit hooks against all files of Pelorus. Then, change a file and add it and commit it to see it in action.

If one of the hooks fails (defined in `.pre-commit-config.yaml` file), the commit is aborted. This can be bypassed by running `git commit` command with the `-n/--no-verify` flag.

### CODE REVIEW

**Comment by KevinMGranger** [comment]
todo for me: check if shellcheck in pre-commit works in macOS

File: .github/workflows/pre-commit-test.yml [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+      - '.pre-commit-config.yaml'
+      - '.github/workflows/pre-commit-test.yml'
+      - 'Makefile'
+
+jobs:
+  deploy-documentation:
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ubuntu-latest, macos-latest, windows-latest]
```
Comment (mateusoliveira43) [inline .github/workflows/pre-commit-test.yml:21]: remove windows

**Comment by KevinMGranger** [comment]
Since the `pre-run` stuff is only necessary for CI (and makes assumptions about the user's repo), maybe it should be part of the github CI files only?

**Comment by mateusoliveira43** [comment]
> Since the `pre-run` stuff is only necessary for CI (and makes assumptions about the user's repo), maybe it should be part of the github CI files only?

this is to address #692 not the CI

File: .github/workflows/pre-commit-test.yml [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+  pre-commit-test:
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ubuntu-latest, macos-latest]
+    runs-on: ${{ matrix.os }}
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+        # with:
```
Comment (mateusoliveira43) [inline .github/workflows/pre-commit-test.yml:26]: remove comment

**Review by mpryc — APPROVED** [review]

**Pull request merged**